Add "In the Zone" indicator

axl13 opened the pull request.

axl13 committed:
Add "In the Zone" indicator

axl13 committed:
Improve readability. Rearrange properties by usefulness.

axl13 committed:
Colorfest

xopok commented on source/PowerAdjusterView.mc:
```diff
@@ -161,6 +160,13 @@ class PowerDataField extends Ui.DataField {
+      var max_power = 900;  // just the upper level
+      if (power < 0 ) { return 0; }
+      if (zone < my_zones.size()){ max_power = my_zones[zone]; }
+      return (power - my_zones[zone-1]).toFloat()/(max_power - my_zones[zone-1]);
```
Деление на 0 может быть, если зоны в настройках заканчиваются на 900, и текущая моща >900.

axl13 replied:
Changed to Last zone + 200

xopok replied:
Forgot to push changes?

xopok commented on source/PowerAdjusterView.mc:
```diff
@@ -227,17 +233,43 @@ class PowerDataField extends Ui.DataField {
+      drawZones(dc, zone, powerValue);
       var l = Ui.View.findDrawableById("label");
-      l.setText(label);
+      l.setText(label + " z:" + zone);
```
: не нужно, у часов и так экран маленький.

axl13 replied:
done

xopok reviewed (commented):
Попозже причешу, сабмить пока так.

axl13 committed:
In no zones, or too few set - display power only.
Some bugfixing

axl13 committed:
Bug fixing

xopok commented on source/PowerAdjusterView.mc:
```diff
@@ -255,12 +258,11 @@ class PowerDataField extends Ui.DataField {
 	  var l = Ui.View.findDrawableById("label");
       if (my_zones.size() > 2) {
+     	  dc.setColor(Graphics.COLOR_WHITE, Graphics.COLOR_BLACK);
           l.setColor(Gfx.COLOR_WHITE);
```
https://developer.garmin.com/downloads/connect-iq/monkey-c/doc/Toybox/WatchUi/DataField.html#getBackgroundColor-instance_method
This can be used to determine if the rest of the watch UI is white on black or black on white.

xopok commented on resources/resources.xml:
```diff
@@ -6,8 +6,11 @@
 				justification="Graphics.TEXT_JUSTIFY_CENTER"/>
-		</layout>
+		  <label id="mark" x="center" y="top"
+				font="Graphics.FONT_SYSTEM_XTINY"
```
Default is FONT_TINY for label and FONT_LARGE for value. Otherwise this datafield stands out from the rest of the UI.
Also on the watch the label sits above the value, seems there should be multiple layouts... E.g. https://github.com/creacominc/connectiq-PowerField/blob/master/resources/layouts/layouts.xml

axl13 committed:
Improve readability of the power numbers

Properly handle background colors when changed.
Remove devices with no native BikeListener

axl13 committed:
Merge branch 'master' into PurePower
